### PR TITLE
PDCL-1305: Create uniqueItems() validator for array validation

### PR DESCRIPTION
**Pull request by adityaguruprasad**
## Description
Creating a uniqueItems() validator for array validation

Use-cases: getIdentity and decisionScope
- getIdentity: array validator validates that edge cases such as ["ECID", "ECID"] aren't allowed to pass
- decisionScope: array validator validates that decisionScope array values cannot contain duplicates

## Related Issue
Currently no array unique items validator in place to ensure that duplicate values aren't passed on command through getIdentity or decisionScope.

## Motivation and Context
Ensuring that all array values passed in getIdentity and decisionScope are unique.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [x] I have made any necessary test changes and all tests pass.
- [x] I have run the Sandbox successfully.

### CODE REVIEW

File: src/utils/isUnique.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+/**
+ * Returns whether an array contains unique values.
+ * @param {*} value
+ * @returns {boolean}
+ */
+export default value => new Set(value).size === Array(...value).length;
```
Comment (jonsnyder) [inline src/utils/isUnique.js:17]: This doesn't work in Internet Explorer 11. That's why the tests are failing. `new Set(iterable)` is not supported, you can create a new set and add all the items to it.

**Review by jonsnyder — CHANGES_REQUESTED** [review]

**Review by jonsnyder — APPROVED** [review]

File: src/utils/isUnique.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    const item = value[i];
+    if (item in storedVals) {
+      return false;
+    }
+    storedVals[item] = true;
+  }
+  return true;
+};
+
+// new Set(value).size === Array(...value).length;
```
Comment (jfkhoury) [inline src/utils/isUnique.js:29]: Let's get rid of this comment.

File: src/utils/isUnique.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+/**
+ * Returns whether an array contains unique values.
+ * @param {*} value
+ * @returns {boolean}
+ */
+export default value => {
```
Comment (jfkhoury) [inline src/utils/isUnique.js:17]: Should we rename `value` to `values`?

File: src/utils/validation/createUniqueItemsValidator.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+import assertValid from "./assertValid";
+import isUnique from "../isUnique";
+
+export default () => {
+  // const values = [];
```
Comment (jfkhoury) [inline src/utils/validation/createUniqueItemsValidator.js:16]: Let's get rid of this comment.

**Review by jfkhoury — APPROVED** [review]
Couple of quick comments, otherwise great stuff.

**Pull request merged**